### DISCUSSION TITLE
Avalonia does not respect dark theme preference on Linux

### DISCUSSION THREAD

**Discussion by TheArcaneBrony** [AvaloniaUI/Avalonia#13994] [Q&A]
## Describe the bug
Avalonia does not respect dark theme preference, unlike other apps

## To Reproduce
`dotnet new avalonia.app`
`dotnet run`

## Expected behavior
See "Welcome to Avalonia!" in dark theme

## Screenshots
I am unable to upload files to GitHub's CDN, so files below are hosted outside of GitHub:
Avalonia app ignoring dark theme preference, displayed on top of Nemo (file browser): https://conduit.rory.gay/_matrix/media/v3/download/conduit.rory.gay/8i3yuutccQY5Zs4dhIBVuGSpLFhQtqxE

## Environment 

- OS: NixOS (Configuration: https://cgit.rory.gay/Rory-Open-Architecture.git, see `Rory-desktop` output)
- Avalonia-Version: 11.0.6 (can reproduce on nightly)

## Additional context
NixOS configuration can be reproduced, if desired.

**Comment by maxkatz6** [comment]
@TheArcaneBrony 
1. What DBus debugging tools are reporting for `org.freedesktop.appearance` settings? For example, you can check with `D-feet`.
2. Have you tested with nightly builds?

**Comment by timunie** [comment]
@maxkatz6 nightly was tested in telegram chat already. It didn't help either acc. to feedback. Looks like it's a custom Linux build so don't know how to debug it properly. But we can get that Linux build on request if I understood correctly

**Comment by maxkatz6** [comment]
@timunie if `org.freedesktop.appearance` doesn't return correct values on a custom Linux build, then this issue should be closed.

**Comment by timunie** [comment]
@TheArcaneBrony can you help us and post what `org.freedesktop.appearance` returns on your system? 

> [!NOTE]
> As we think it may be related to your custom system I'll make this a discussion for now. If we later find the root cause we can still decide to make it a feature request or a bug report again. (Depends on the input we can get here)

**Comment by TheArcaneBrony** [comment]
org.freedesktop.appearance does not return anything, as not a single desktop portal implementation i've tried works at all (they all NOOP due to not being in their respective desktop environment, which doesn't work on i3)
This also happens on nightly builds.
I have set dark mode preference in DConf, which most apps respect just fine.

**Comment by TheArcaneBrony** [comment]
For what it's worth, here's a screenshot of D-feet with a desktop portal NOOP-ing.
This behavior is exactly the same with xdg-desktop-portal-gtk, as with any other desktop portal implementation I have tried.
![Screenshot from 2023-12-19 20-08-27](https://github.com/AvaloniaUI/Avalonia/assets/13570458/867c1b0f-9793-4cc3-be76-af73a6119eac)

**Reply by maxkatz6** [reply]
I don't have Linux near me, but I have WSL2 installed with some GTK apps in it, which seems to be enough.
![image](https://github.com/AvaloniaUI/Avalonia/assets/3163374/7ac9449a-e70b-43b6-8056-21b8c9761700)

![image](https://github.com/AvaloniaUI/Avalonia/assets/3163374/d8b9c0eb-6a26-43a5-a328-fa3a476f2c73)

**Reply by maxkatz6** [reply]
For some reason GTK apps themselves on WSL (including WSL window frame) don't respect org.freedesktop.appearance settings, but that's not ours problem at this point.

**Reply by TheArcaneBrony** [reply]
WSL window frame doesnt follow any settings
as for those properties...
![Screenshot from 2023-12-20 06-20-01](https://github.com/AvaloniaUI/Avalonia/assets/13570458/8759ae77-7a91-4093-9908-1e1a54c2361f)
![Screenshot from 2023-12-20 06-21-37](https://github.com/AvaloniaUI/Avalonia/assets/13570458/6a6a3bfc-9e4d-4be0-a9cd-4c1f85d6a9b9)

**Reply by TheArcaneBrony** [reply]
I feel like there should be some kind of fallback? 
eg. ~/.config/gtk-3.0/settings.ini contains this:
```toml
[Settings]
gtk-application-prefer-dark-theme = true
```

**Reply by maxkatz6** [reply]
Linux has unnumerable number of ways how to configure a single thing, sometimes more than one per desktop environment. 
XDG is a standard, more or less, which Avalonia handles. I am not sure if we should read GTK specific config files.

**Reply by workgroupengineering** [reply]
We can add a developer-customizable fallback callback.

**Reply by TheArcaneBrony** [reply]
> Linux has unnumerable number of ways how to configure a single thing, sometimes more than one per desktop environment. XDG is a standard, more or less, which Avalonia handles. I am not sure if we should read GTK specific config files.

... as you demonstrated by using *gsettings* to set org.***gnome***.desktop.interface options, nice standardisation.

In any case, I shouldn't have to add a custom environment variable to override requested variant myself, right?
And I don't think Qt reads that file, yet Qt uses dark theme.

**Reply by timunie** [reply]
> In any case, I shouldn't have to add a custom environment variable to override requested variant myself, right? 

You should be able to read that setting on your own as a fallback, yes. But that is maybe up to you to do it on startup. 

```cs
if (OperatingSystem.IsLinux())
{
    // ...
}
```

**Comment by TheArcaneBrony** [comment]
In addition, NixOS also isn't really "custom", it just takes a completely different approach at package management, hence I'd linked my configuration so it can be reproduced or looked over.

**Reply by maxkatz6** [reply]
Could be the reason https://github.com/NixOS/nixpkgs/issues/202810 https://github.com/NixOS/nixpkgs/issues/202810#issuecomment-1532008182

**Reply by TheArcaneBrony** [reply]
nope, thats a KDE bug (not that restarting the desktop portal does anything)

**Comment by CabdiRaxmaan-7** [comment]
13999
